timesketch/tsctl.py

```diff
@@ -1351,3 +1363,579 @@ def list_config():
+                    sketch, preventing event fetching.
+        # Note: Other exceptions from the datastore interaction or file reading
+        # are caught and logged, resulting in an empty string return value.
+    """
+    print("Exporting events...")
+    query_string = "*"
+    query_filter = {
+        "indices": "_all",
+        "size": 10000,  # Use default size, export handles scrolling
+    }
+    query_dsl = None
+
+    indices, _ = lib_utils.get_validated_indices("_all", sketch)
+    if not indices:
+        print("WARNING: No valid indices found via primary method, trying fallback.")
+        indices = [t.searchindex.index_name for t in sketch.active_timelines]
+        if not indices:
+            raise ValueError(
+                "ERROR: No active timelines (and thus no indices) found"
+                "for this sketch."
+            )
+
+    print("  Requesting event data (preferring JSONL)...")
+    event_file_handle = api_export.query_to_filehandle(
+        query_string=query_string,
+        query_filter=query_filter,
+        query_dsl=query_dsl,
+        indices=indices,
+        sketch=sketch,
+        datastore=datastore,
+        return_fields=return_fields,
+    )
+    event_file_handle.seek(0)
+
+    try:
+        content_str = event_file_handle.read()
+        if not isinstance(content_str, str):
+            content_str = content_str.decode("utf-8", errors="replace")
+    except Exception as read_err:  # pylint: disable=broad-except
+        print(f"  ERROR reading event data stream: {read_err}")
+        content_str = ""
+
+    input_content = content_str.strip()
+    is_likely_jsonl = False
+    if input_content and input_content.startswith("{"):
+        try:
+            first_line = input_content.split("\n", 1)[0]
+            json.loads(first_line)
+            is_likely_jsonl = True
+            print("  Detected JSONL format in response.")
+        except Exception:  # pylint: disable=broad-except
+            is_likely_jsonl = False
+            print("  Detected non-JSONL format in response (assuming CSV).")
+    elif input_content:
+        print("  Detected non-JSONL format in response (assuming CSV).")
+    else:
+        print("  WARNING: Received empty content from export function.")
+
+    return input_content, is_likely_jsonl
+
+
+# Helper function to convert event data
+def _convert_event_data(
+    input_content: str,
+    is_likely_jsonl: bool,
+    output_format: str,
+    return_fields: list,
+) -> bytes:
+    """Converts fetched event data between JSONL and CSV formats.
+    This function takes the raw event data string, determines the necessary
+    conversion based on the detected input format (`is_likely_jsonl`) and the
+    desired `output_format`, and performs the conversion.
+    - If the input is JSONL and the output is CSV, it parses each JSON line,
+      extracts relevant fields, handles list values (joining with '|'), and
```

Comment:
> ... list values (joining with '|') ...

Why not represent lists as "[value1, value2]" string? This is also what we use to represent list in the Timesketch frontend.

Reply:
Also, currently empty lists like `tag` are presented as empty field in CSV. Maybe a "[]" is clearer for people to understand?